batch/batch/batch.py

```diff
@@ -1234,10 +1212,34 @@ def is_successful(self):
         await job.mark_unscheduled()
         return
 
-    if pod and pod.status and pod.status.phase in ('Succeeded', 'Failed'):
-        log.info(f'job {job.id} mark complete')
-        await job.mark_complete(pod)
-        return
+    if pod and pod.status:
+        if pod.status.init_container_statuses:
+            assert len(pod.status.init_container_statuses) == 1
+            init_status = pod.status.init_container_statuses[0]
+            if init_status.state and init_status.state.terminated and init_status.state.terminated.exit_code != 0:
+                log.info(f'job {job.id} mark unscheduled -- setup container exited {init_status.state.terminated.exit_code}')
+                await job.mark_unscheduled()
```

Comment:
I'm uncomfortable with this. What if the input files to copy don't exist? We'll keep trying to copy over and over again.

Reply:
OK, I changed it to `mark_failed`. This was part of my attempts to avoid failing when `gsutil` failed to authorize correctly, which isn't really a user error, it's a GCP error that we propagate up. Hopefully the other changes I made to retry `gcloud auth` are enough.

Reply:
The best thing to do would be to distinguish user errors from other errors and retry if its a non-user error.